### PR TITLE
Check react native prerequisites

### PR DESCRIPTION
**Pull request by dgomezc**
## PR checklist

- Quick summary of changes
  - Get platform requirement in extension code and send to client
  - Save platform requirements in redux store
  - Show notification in right sidebar
  - Add modal to show platform requirements
  - Add link in bottom of rigth sidebar
- Which issue does this PR relate to?
   - #1665 - React Native: Check and warn about missing RNW pre-requisites

### CODE REVIEW

File: src/client/src/App.tsx [hunk 1]
```diff
@@ -35,6 +35,9 @@ const ViewLicensesModal = loadable(
 const AzureServicesModal = loadable(
   () => import(/* webpackChunkName: "AzureServicesModal" */ "./modals/AzureServicesModal")
 );
+const ViewPlatformRequirementsModal = loadable(
```
Comment (prupipho) [inline src/client/src/App.tsx]: We could remove "View" from the name to shorten it a bit I think.

File: src/client/src/components/RightSidebar/index.tsx [hunk 2]
```diff
@@ -80,9 +88,21 @@ const RightSidebar = (props: Props) => {
...
 
-            <div className={styles.marginLeft}>
-              <div className={styles.title}>{formatMessage(messages.yourProjectDetails)}</div>
+            {hasInvalidRequirements && (
+              <div className={styles.notificationContainer}>
+                <Notification showWarning={true} altMessage={formatMessage(messages.missingRequirements)}>
+                  {formatMessage(messages.missingRequirements)}
+                  <button
+                    className={classnames(styles.notificationLink, buttonStyles.buttonLink)}
+                    onClick={() => dispatch(ModalActions.openViewPlatformRequirementsAction())}
```
Comment (prupipho) [inline src/client/src/components/RightSidebar/index.tsx]: if we decide to change name we should probably remove "View" from here too

File: src/client/src/components/RightSidebar/index.tsx [hunk 3]
```diff
@@ -95,13 +115,22 @@ const RightSidebar = (props: Props) => {
...
                     onClick={() => dispatch(ModalActions.openViewLicensesModalAction())}
                   >
                     {formatMessage(messages.viewLicenses)}
                   </button>
                 </div>
               )}
+              {hasRequirements && (
+                <button
+                  className={buttonStyles.buttonLink}
+                  onClick={() => dispatch(ModalActions.openViewPlatformRequirementsAction())}
```
Comment (prupipho) [inline src/client/src/components/RightSidebar/index.tsx]: same here remove "View" if we rename the previous one.

File: src/client/src/mockData/extensionModules/mockCoreTSModule.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@ import * as mockWebData from "./mockData/mockWebPlatformData";
 import * as mockReactNativeData from "./mockData/mockReactNativePlatformData";
 
 //DEV:Change mock data to change platform
-const mockData = mockWebData;
+const mockData = mockReactNativeData;
```
Comment (prupipho) [inline src/client/src/mockData/extensionModules/mockCoreTSModule.ts]: better commit this with the Web mock instead?

File: src/client/src/modals/ViewPlatformRequirementsModal/RequirementItem/messages.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { defineMessages } from "react-intl";
+
+const messages = defineMessages({
+  requirementInstalled: {
+    id: "ViewPlatformRequirementsModal.requirementItem.requirementInstalled",
```
Comment (prupipho) [inline src/client/src/modals/ViewPlatformRequirementsModal/RequirementItem/messages.ts]: again, only if we decide renaming, remove the "View"

File: src/client/src/modals/ViewPlatformRequirementsModal/messages.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+const messages = defineMessages({
+  developmentRequirements: {
+    id: "ViewPlatformRequirementsModal.developmentRequirements",
+    defaultMessage: "Development Requirements",
+  },
+  ariaCloseModalLabel: {
+    id: "ViewPlatformRequirementsModal.ariaCloseModalLabel",
+    defaultMessage: "Close Platform Requirements Modal",
+  },
+  checkAndInstallRequirements: {
```
Comment (prupipho) [inline src/client/src/modals/ViewPlatformRequirementsModal/messages.ts]: Not sure about those 2 resources... maybe it´s better having just one 
`"To check and install missing depevelopment requirements {0}."`
and the other one 
`"review the docs"`

And then doing the replace **{0}** by the `<a... {review the docs } ... >`

File: src/extension/src/scripts/rnw-dependencies.ps1 [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+    }
+}
+
+
+if ($Installed -ne 0) {
+    Write-Output "Installed $Installed dependencies. You may need to close this window for changes to take effect."
+}
+
+if ($NeedsRerun -ne 0) {
+    #TODO: Set comment
```
Comment (prupipho) [inline src/extension/src/scripts/rnw-dependencies.ps1]: Is this where we said we may add a WTS COMMENT: we commented this to be able to return info blablablah? :)

File: src/client/src/components/RightSidebar/messages.ts [hunk 8]
```diff
@@ -21,11 +21,22 @@ const messages = defineMessages({
...
+    defaultMessage: "Missing requirements.",
+  },
+  viewDetails: {
+    id: "rightSidebar.viewDetails",
+    defaultMessage: " View details",
+  },
+  viewRequirements: {
+    id: "rightSidebar.viewRequirements",
+    defaultMessage: " View requirements",
+  },
```
Comment (prupipho) [inline src/client/src/components/RightSidebar/messages.ts]:
```suggestion
  missingRequirements: {
    id: "rightSidebar.missingRequirements",
    defaultMessage: "Missing requirements.",
  },
  viewDetails: {
    id: "rightSidebar.viewDetails",
    defaultMessage: "View details",
  },
  viewRequirements: {
    id: "rightSidebar.viewRequirements",
    defaultMessage: "View requirements",
  },
```

**Comment by prupipho** [comment]
Review and notify that the prerequisites script is only run in Windows. If not just send a link to the react native windows site.

File: src/client/src/components/RightSidebar/About/messages.ts [hunk 9]
```diff
@@ -1,6 +1,10 @@
 import { defineMessages } from "react-intl";
 
 const messages = defineMessages({
+  title: {
+    id: "rightsidebar.about.title",
+    defaultMessage: "About",
```
Comment (prupipho) [inline src/client/src/components/RightSidebar/About/messages.ts:6]: Change case in "Report an issue"

File: src/client/src/App.tsx (hunk 1 above)
Comment (dgomezc) [reply src/client/src/App.tsx]: Done.

File: src/client/src/components/RightSidebar/index.tsx (hunk 2 above)
Comment (dgomezc) [reply src/client/src/components/RightSidebar/index.tsx]: Done.

File: src/client/src/components/RightSidebar/index.tsx (hunk 3 above)
Comment (dgomezc) [reply src/client/src/components/RightSidebar/index.tsx]: Done.

File: src/client/src/mockData/extensionModules/mockCoreTSModule.ts (hunk 4 above)
Comment (dgomezc) [reply src/client/src/mockData/extensionModules/mockCoreTSModule.ts]: Done.

File: src/client/src/modals/ViewPlatformRequirementsModal/RequirementItem/messages.ts (hunk 5 above)
Comment (dgomezc) [reply src/client/src/modals/ViewPlatformRequirementsModal/RequirementItem/messages.ts]: Done.

File: src/extension/src/scripts/rnw-dependencies.ps1 (hunk 7 above)
Comment (dgomezc) [reply src/extension/src/scripts/rnw-dependencies.ps1]: Done.

File: src/client/src/components/RightSidebar/messages.ts (hunk 8 above)
Comment (dgomezc) [reply src/client/src/components/RightSidebar/messages.ts]: Done.

File: src/client/src/modals/ViewPlatformRequirementsModal/messages.ts (hunk 6 above)
Comment (dgomezc) [reply src/client/src/modals/ViewPlatformRequirementsModal/messages.ts]: Split in text and bottom link.

**Review by prupipho — APPROVED** [review]

**Pull request merged**